db/db_impl.cc

```diff
@@ -1352,6 +1352,15 @@ ColumnFamilyHandle* DBImpl::DefaultColumnFamily() const {
 Status DBImpl::Get(const ReadOptions& read_options,
                    ColumnFamilyHandle* column_family, const Slice& key,
                    PinnableSlice* value) {
+  if (read_options.timestamp != nullptr) {
```

Comment:
Maybe not in this PR, but we should think about returning which timestamp for the entry returned.

Reply:
Do you mean we return the timestamp to user? Say if user specifies 55 as timestamp, and the db has a record with ts 54, we need to return 54 as well as the key-value pair?